modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts

```diff
@@ -394,8 +401,67 @@ export class EddsaUtils extends baseTSSUtils<KeyShare> {
     const signablePayload = Buffer.from(unsignedTx.signableHex, 'hex');
 
     const userSignShare = await createUserSignShare(signablePayload, signingKey.pShare);
+    const commitment = userSignShare.rShares[bitgoIndex]?.commitment;
+    assert(commitment, 'Unable to find commitment in userSignShare');
+    const userToBitgoCommitment = this.createUserToBitgoCommitmentShare(commitment);
+
+    const signerShare = signingKey.yShares[bitgoIndex].u + signingKey.yShares[bitgoIndex].chaincode;
+    const bitgoGpgKey = await getBitgoGpgPubKey(this.bitgo);
+    const bitgoToUserEncryptedSignerShare = await encryptText(signerShare, bitgoGpgKey);
 
-    return { rShare: userSignShare, signingKeyYShare: signingKey.yShares[3] };
+    const encryptedSignerShare = this.createUserToBitgoEncryptedSignerShare(bitgoToUserEncryptedSignerShare);
+    const stringifiedRShare = JSON.stringify(userSignShare);
+    const encryptedRShare = this.bitgo.encrypt({ input: stringifiedRShare, password: params.walletPassphrase });
+    const encryptedUserToBitgoRShare = this.createUserToBitgoEncryptedRShare(encryptedRShare);
+
+    return { userToBitgoCommitment, encryptedSignerShare, encryptedUserToBitgoRShare };
+  }
+
+  async createRShareFromTxRequest(params: {
+    txRequest: TxRequest;
+    prv: string;
+    walletPassphrase?: string;
+    encryptedUserToBitgoRShare?: EncryptedSignerShareRecord;
+  }): Promise<{ rShare: SignShare; signingKeyYShare: YShare }> {
+    const { txRequest, prv, walletPassphrase, encryptedUserToBitgoRShare } = params;
+    const txRequestResolved: TxRequest = txRequest;
+
+    const hdTree = await Ed25519BIP32.initialize();
+    const MPC = await Eddsa.initialize(hdTree);
+
+    const userSigningMaterial: SigningMaterial = JSON.parse(prv);
+    if (!userSigningMaterial.backupYShare) {
+      throw new Error('Invalid user key - missing backupYShare');
+    }
+
+    assert(txRequestResolved.transactions || txRequestResolved.unsignedTxs, 'Unable to find transactions in txRequest');
+    const unsignedTx =
+      txRequestResolved.apiVersion === 'full'
+        ? txRequestResolved.transactions![0].unsignedTx
+        : txRequestResolved.unsignedTxs[0];
+
+    const signingKey = MPC.keyDerive(
+      userSigningMaterial.uShare,
+      [userSigningMaterial.bitgoYShare, userSigningMaterial.backupYShare],
+      unsignedTx.derivationPath
+    );
+
+    let rShare: SignShare;
+    if (walletPassphrase && encryptedUserToBitgoRShare) {
```

Comment:
Do we need to worry about backwards compatibility here? No, right? Since we control the full stack, i.e. both instances of express, iIthink this path wouldnt need to produce signingKeyYShare to BitGo. As that would have been returned by the previous call and already used by BitGo in the first commitment round. So, technically, we could do without this if condition and always use a flow that only returns the user's R share at this decommitment step. As a consequence, we wouldnt even need the prv key for this step, just the wallet password.

Reply:
Hmm, maybe we can remove the compatibility later, after we enforce it. The problem with making a breaking change now, is that if something else needs to be released too they gonna need to upgrade the express version and hiting this breaking change.

Reply:
But what exactly is it breaking? Once a client upgrades express they get both express (external signer) and express (platform) working well together since the new flow would be available to both instances. And clients with older express versions should still be fine due to backwards compatibility support within our stack for now.

Reply:
The clients may want to upgrade express to bring other changes and not fix this if we make a breaking change.

Reply:
but that upgrade will bring in all your new changes with them that include commitment and the whole thing would still work no? Clients dont have the option to upgrade only certain parts of express, and there is no reason a client should not want to have commitments included.